lib/buildFactory.js

```diff
@@ -0,0 +1,147 @@
+ * @return {Promise}
+ */
+function getCommitSha(datastore, config) {
+    if (config.sha) {
+        return new Promise((resolve) => resolve(config.sha));
+    }
+
+    const jobFactory = new JobFactory({ datastore });
+    const userFactory = new UserFactory({ datastore });
+
+    return Promise.all([
+        jobFactory.get(config.jobId),
+        userFactory.get({ username: config.username })
+    ]).then(models => {
+        const job = models[0];
+        const user = models[1];
+
+        return job.pipeline
+            .then(pipeline => githubHelper.getInfo(pipeline.scmUrl))
+            .then(repoInfo => githubHelper.run({
+                user,
+                action: 'getBranch',
+                params: repoInfo
+            }))
+            .then(data => data.commit.sha);
+    });
+}
+
+class BuildFactory extends BaseFactory {
+    /**
+     * Construct a JobFactory object
+     * @method constructor
+     * @param  {Object}    config
+     * @param  {Object}    config.datastore         Object that will perform operations on the datastore
+     */
+    constructor(config) {
+        super('build', config);
+    }
+
+    /**
+     * Instantiate a Build class
+     * @method createClass
+     * @param  {Object}     config               Build data
+     * @param  {Datastore}  config.datastore     Datastore instance
+     * @param  {String}     config.id            unique id
+     * @param  {String}     config.username      The user that created this build
+     * @param  {String}     config.jobId         The ID of the associated job to start
+     * @param  {String}     [config.sha]         The sha of the build
+     * @param  {String}     [config.container]   The kind of container to use
+     * @return {Build}
+     */
+    createClass(config) {
+        return new Build(config);
+    }
+
+    /**
+     * Create a new Build
+     * @method create
+     * @param  {Object}   config                Config object
+     * @param  {String}   config.username       The user that created this build
+     * @param  {String}   config.jobId          The ID of the associated job to start
+     * @param  {String}   config.executor       Build executor instance
+     * @param  {String}   [config.sha]          The sha of the build
+     * @param  {String}   [config.container]    The kind of container to use
+     * @return {Promise}
+     */
+    create(config) {
+        const container = config.container || 'node:4';
+        const now = Date.now();
+        const modelConfig = hoek.applyToDefaults({
+            cause: `Started by user ${config.username}`,
+            container,
+            createTime: now,
+            number: now,
+            status: 'QUEUED'
+        }, config);
+
+        return getCommitSha(this.datastore, modelConfig)
+            .then(sha => {
+                modelConfig.sha = sha;
+
+                return super.create(modelConfig);
+            })
+            .then(build =>
+                build.start()
```

Comment:
If executor is passed as part of `create`, `build.start()` should work just fine.